Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,248 @@
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE
+#  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+#  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+#  HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+#  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+#  OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+#  SUCH DAMAGE.
+                       
+
+param (
+    [switch]$IncludeSensitiveData
+)
+
+function Get-SystemConfiguration {
+    try {
+        Start-Process -FilePath "msinfo32.exe" -ArgumentList "/report", "msinfo32.txt" -Wait
+    } catch {
+        Write-Warning "Failed to collect system configuration: $_"
+    }
+}
+
+function Get-EventLogs {
+    try {
+        $logTypes = @("system", "security", "application")
+        foreach ($logType in $logTypes) {
+            wevtutil epl $logType "$logType.evtx"
+            wevtutil al "$logType.evtx"
+        }
+    } catch {
+        Write-Warning "Failed to collect event logs: $_"
+    }
+}
+
+function Get-DriversList {
+    if (-not ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
+        Write-Warning "This function requires administrative privileges. Please run the script as an administrator."
+        return
+    }
+    try {
+        $drivers = Get-WindowsDriver -Online -All
+        if ($drivers) {
+            $drivers | Select-Object -Property * | Export-Csv -Path drv_list.csv -NoTypeInformation
+        } else {
+            Write-Warning "No drivers found."
+        }
+    } catch {
+        Write-Warning "Failed to collect drivers list: $_"
+    }
+}
+
+function Get-VirtioWinStorageDrivers {
+    $registryPaths = @(
+        "HKLM:\SYSTEM\CurrentControlSet\Services\Disk",
+        "HKLM:\SYSTEM\CurrentControlSet\Services\viostor\Parameters",
+        "HKLM:\SYSTEM\CurrentControlSet\Services\vioscsi\Parameters"
+    )
+    $valuesToQuery = @("IoTimeoutValue", "TimeoutValue")
+
+    foreach ($path in $registryPaths) {
+        foreach ($value in $valuesToQuery) {
+            try {
+                $property = Get-ItemProperty -Path $path -Name $value -ErrorAction Stop
+                $output = "$path\$value : $($property.$value)"
+                $output | Out-File -FilePath virtio_disk.txt -Append
+            } catch {
+                Write-Warning "Failed to collect ${path}\${value}: $_"
+            }
+        }
+    }
+}
+
+function Get-WindowsUpdateLogs {
+    try {
+        $logPath = Join-Path -Path (Get-Location) -ChildPath "WindowsUpdate.log"
+        if (Get-Command -Name Get-WindowsUpdateLog -ErrorAction SilentlyContinue) {
```

Comment:
Is this check really necessary? Won't `Start-Process` throw when it doesn't exist?

Reply:
even if you want to check that `Get-WindowsUpdateLog` is present, why do you need to call extra PowerShell to run the command?

Reply:
> even if you want to check that `Get-WindowsUpdateLog` is present, why do you need to call extra PowerShell to run the command?

Cause Get-WindowsUpdateLog
genetate some annoing output 
`Converting C:\Windows\logs\WindowsUpdate into C:\Users\Jedoku\Desktop\collector\kvm-guest-drivers-windows\Tools\debug\WindowsUpdate.log ...

Input
----------------
File(s):
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.114705.284.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.140120.096.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.151337.033.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.164451.571.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.164451.571.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.223755.522.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.223755.522.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.230928.543.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.232534.520.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.233045.478.1.etl

0.00%7.96%15.92%23.88%31.84%39.80%47.76%55.72%63.68%71.64%79.60%87.56%95.52%100.00%

Output
----------------
DumpFile:           C:\Users\Jedoku\AppData\Local\Temp\WindowsUpdateLog\wuetl.CSV.tmp.00000

The command completed successfully.















Input
----------------
File(s):
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240603.233045.478.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240604.104828.379.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240604.104828.379.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240604.113138.705.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240604.114831.921.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240604.151321.122.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.013036.555.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.013036.555.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.013036.555.3.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.111238.225.1.etl

0.00%7.51%15.02%22.54%30.05%37.56%45.07%52.58%60.09%67.61%75.12%82.63%90.14%97.65%100.00%

Output
----------------
DumpFile:           C:\Users\Jedoku\AppData\Local\Temp\WindowsUpdateLog\wuetl.CSV.tmp.00001

The command completed successfully.
















Input
----------------
File(s):
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.111238.225.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.111238.225.3.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.151409.034.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.212615.385.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240605.212615.385.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240606.094859.313.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240606.094859.313.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240606.101528.666.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240606.101528.666.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240606.101919.933.1.etl

0.00%7.02%14.04%21.05%28.07%35.09%42.11%49.12%56.14%63.16%70.18%77.19%84.21%91.23%98.25%100.00%

Output
----------------
DumpFile:           C:\Users\Jedoku\AppData\Local\Temp\WindowsUpdateLog\wuetl.CSV.tmp.00002

The command completed successfully.

















Input
----------------
File(s):
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240606.153739.054.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240606.153739.054.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240607.193000.072.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240607.193000.072.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240607.193000.072.3.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.095902.694.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.095902.694.2.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.095902.694.3.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.105237.888.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.192837.111.1.etl

0.00%7.41%14.81%22.22%29.63%37.04%44.44%51.85%59.26%66.67%74.07%81.48%88.89%96.30%100.00%

Output
----------------
DumpFile:           C:\Users\Jedoku\AppData\Local\Temp\WindowsUpdateLog\wuetl.CSV.tmp.00003

The command completed successfully.
















Input
----------------
File(s):
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.194115.185.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.212115.224.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240608.234440.831.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.032845.856.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.055903.458.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.100802.105.1.etl                                                   C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.130917.993.1.etl                                                   C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.130917.993.2.etl                                                   C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.192840.039.1.etl                                                   C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.203152.987.1.etl                                                                                                                                                                      0.00%13.68%27.35%41.03%54.70%68.38%82.05%95.73%100.00%

Output
----------------
DumpFile:           C:\Users\Jedoku\AppData\Local\Temp\WindowsUpdateLog\wuetl.CSV.tmp.00004

The command completed successfully.










Input
----------------
File(s):
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240609.222115.255.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240610.034548.024.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240610.055902.479.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240610.084641.982.1.etl
     C:\Windows\logs\WindowsUpdate\WindowsUpdate.20240610.100803.881.1.etl

0.00%47.06%94.12%100.00%

Output
----------------
DumpFile:           C:\Users\Jedoku\AppData\Local\Temp\WindowsUpdateLog\wuetl.CSV.tmp.00005

The command completed successfully.





WindowsUpdate.log written to C:\Users\Jedoku\Desktop\collector\kvm-guest-drivers-windows\Tools\debug\WindowsUpdate.log
`

Reply:
@YanVugenfirer Is it a problem to have extra output? 

@Jedoku Solution without spawning new PowerShell
```powershell
$Job = Start-Job -ScriptBlock {Get-WindowsUpdateLog}
$Job | Wait-Job | Remove-Job
```

Reply:
It seems Powershell supports the POSIX redirection pattern so use it: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_redirection?view=powershell-7.4